nexus/src/db/model.rs

```diff
@@ -101,6 +101,67 @@ macro_rules! impl_enum_type {
     }
 }
 
+macro_rules! impl_enum_type2 {
+    (
+        $(#[$enum_meta:meta])*
+        pub struct $diesel_type:ident;
+
+        $(#[$model_meta:meta])*
+        pub enum $model_type:ident;
+        $($enum_item:ident => $sql_value:literal)+
+    ) => {
+
+        $(#[$enum_meta])*
+        pub struct $diesel_type;
+
+        $(#[$model_meta])*
+        pub enum $model_type {
+            $(
+                $enum_item,
+            )*
+        }
```

Comment:
This is the analogous line in the original. Instead of defining the enum, it wrapped the imported external one.

```rs
pub struct $model_type(pub $ext_type);
```